src/main/java/com/robertx22/mine_and_slash/mechanics/thresholds/datapack/SpendThresholdDef.java

```diff
@@ -0,0 +1,95 @@
+package com.robertx22.mine_and_slash.mechanics.thresholds.datapack;
+
+import com.google.gson.annotations.SerializedName;
+import com.robertx22.mine_and_slash.mechanics.thresholds.DataDrivenSpendThresholdSpec;
+import com.robertx22.mine_and_slash.mechanics.thresholds.SpendThresholdSpec;
+import com.robertx22.mine_and_slash.saveclasses.unit.ResourceType;
+import net.minecraft.server.level.ServerPlayer;
+
+import java.util.*;
+
+public class SpendThresholdDef {
+    public String key;
+    public String resource = "energy";
+    public boolean enabled = true;
+    public int priority = 0;
+
+    public static class Threshold {
+        public String mode = "X_PER_LEVEL";
+        public float value = 0f;
+        @SerializedName("multiply_by_level") public boolean multiplyByLevel = false;
+        @SerializedName("pct_of") public String pctOf; // optional
```

Comment:
let's have only full names. if you would do this, please, change all other `pct_of` variables
```suggestion
        @SerializedName("percent_of") public String percentOf; // optional
```